src/container/resolveContainerProps.js

```diff
@@ -1,14 +1,21 @@
 import objectAssign from "object-assign";
-import type DiodeRootContainer from "./DiodeRootContainer";
 
-export default function resolveContainerProps(
-  response,
-  RootContainer: DiodeRootContainer
-) {
+/*
+  Possible Errors:
+  1. Queries key specified in Diode Container is not the same as query type in Diode Query
+*/
+export default function resolveContainerProps(response, RootContainer) {
   // Convert diode response as root container props
   const { map: queryMap } = RootContainer.query.getContainerQuery();
   const containerProps = Object.keys(queryMap).reduce((props, key) => {
-    props[key] = response[queryMap[key].type];
+    props[key] = response[key];
+
```

Comment:
CMIIW, the purpose of this function is to map `response` to `queries` in Diode.Container. 

There seems to be a requirement in Diode. such that the keys in Diode.Container's queries must be the same as `type` in the query. However, `props[key] = response[queryMap[key].type]` is assuming that the keys in Diode.Container's queries are the same as the `type` in the query. If they are different, there will be no error in this function because `type` is always mapped to `response` anyway.

As a result, there will be an uncaught error on `markFetchAllCache` function which is usually executed afterwards, using the data returned by `resolveContainerProps`. Why not just catch the error here instead?

Reply:
I forgot the exact detail, but as far as I can remember, what we want to achieve is to mimick graphql object shape as close as possible, which unfortunately due to multiple query support inside single component, we can't enforce this easily.

The solution was to matching query type with container key that point to said query. So that the enclosing component can expect that a prop with the same name exist to read that query.

```jsx
const Component = props => <div>{props.propKey.x.y}</div>

const Container = Diode.createContainer(Component, {
  queries: {
    propKey: Diode.createQuery(Query, {
      x: {
        y: null
      }
    }
  }
});